ext/auto_flush.c

```diff
@@ -66,7 +66,12 @@ ZEND_RESULT_CODE ddtrace_flush_tracer(bool force_on_startup, bool collect_cycles
                                 .client_computed_top_level = false,
                                 .client_computed_stats = false,
                         };
-                        ddog_MaybeError send_error = ddog_sidecar_send_trace_v04_shm(&ddtrace_sidecar, ddtrace_sidecar_instance_id, shm, written, &tags);
+                        size_t size_hint = written;
+                        zend_long n_requests = get_global_DD_TRACE_AGENT_FLUSH_AFTER_N_REQUESTS();
+                        if (n_requests) {
+                            size_hint = MAX(get_global_DD_TRACE_AGENT_FLUSH_SIZE() / n_requests, size_hint);
```

Comment:
I'm not too fond of this logic living in the tracer instead of having a flush after N request setting in the side car itself, but on the other hand I'm not sure if that setting would make sense for the general data pipeline.

Reply:
Same here, that's why I did it this way.